### PR TITLE
Trim Manual calculations

### PR DESCRIPTION
**Pull request by New0**
Trim manual calculations before saving form and before sending formula to new parser.

For #3270

### CODE REVIEW

File: classes/forms.php [hunk 1]
```diff
@@ -482,15 +482,19 @@ protected static function add_details( $forms ){
...
+				// option value labels
 				if(!empty($field['config']['option']) && is_array($field['config']['option'])){
 					foreach($field['config']['option'] as &$option){
 						if(!isset($option['value'])){
 							$option['value'] = $option['label'];
 						}
 					}
+				// trim manual calculations
+				} else if(!empty($field["config"]["manual_formula"]) && is_string($field["config"]["manual_formula"])){
+					$field["config"]["manual_formula"] = str_replace( "\r", '', str_replace( "\n", '', str_replace( ' ', '', trim( $field["config"]["manual_formula"] ) ) ) );
```
Comment (Shelob9) [inline classes/forms.php]: @New0 This is a lot, without a test.

Can you please add a method to `Caldera_Forms_Santize` class to do this. And test against possible inputs, so we know it is safe?

Also what about `$str = trim( str_replace( '<br>', '', nl2br($str)  ));` nl2br should convert all line breaks to html br tags, which we can then lift out.

File: classes/forms.php [hunk 2]
```diff
@@ -481,16 +481,20 @@ protected static function add_details( $forms ){
...
+				// option value labels
 				if(!empty($field['config']['option']) && is_array($field['config']['option'])){
 					foreach($field['config']['option'] as &$option){
 						if(!isset($option['value'])){
 							$option['value'] = $option['label'];
 						}
 					}
+				// trim manual calculations
+				} else if(!empty($field["config"]["manual_formula"]) && is_string($field["config"]["manual_formula"])){
+					$field["config"]["manual_formula"] = Caldera_Forms_Sanitize::finish_trim( Caldera_Forms_Sanitize::sanitize_header( $field["config"]["manual_formula"] ) );
```
Comment (Shelob9) [inline classes/forms.php]: This is better 👍

File: tests/test-forms-api.php [hunk 3]
```diff
@@ -205,5 +205,35 @@ public function testUpdateAndGet()
...
+      		'required' 		=>	1,
+			'config'		=>	[
+				'manual_formula'	=> ' 2 + 2\n'
+			]
+		];
+		Caldera_Forms_Forms::save_form($form);
+
+		$form = Caldera_Forms_Forms::get_form($form_id);
+
+		$this->assertSame( '2 + 2', $form['fields']['fld_60011111']['config']['manual_formula'] );
```
Comment (Shelob9) [inline tests/test-forms-api.php]: 🌮

**Comment by New0** [comment]
Travis passed at https://travis-ci.org/CalderaWP/Caldera-Forms/builds/560540426 but doesn't seem to be  returning the status

**Comment by New0** [comment]
Passing build after update https://travis-ci.org/CalderaWP/Caldera-Forms/builds/562101324

**Pull request merged**